### PR TITLE
socketpool: refine pool entry timeout

### PR DESCRIPTION
**Pull request by chenbd**
This PR try calculating the exact timeout for socketpool's pool entry.

### CODE REVIEW

File: lib/common/socketpool.c [hunk 1]
```diff
@@ -82,36 +84,49 @@ static void destroy_attached(struct pool_entry_t *entry)
...
+            struct pool_entry_t *entry = H2O_STRUCT_FROM_MEMBER(struct pool_entry_t, all_link, pool->_shared.sockets.next);
+            if (entry->added_at > expire_before) {
+                return entry->added_at + pool->timeout - now_ms;
+            }
+            destroy_attached(entry);
+            __sync_sub_and_fetch(&pool->_shared.count, 1);
+            __sync_sub_and_fetch(&pool->_shared.pooled_count, 1);
+        }
+    }
+    return SIZE_MAX;
```
Comment (kazuho) [inline lib/common/socketpool.c]: UINT64_MAX?

File: lib/common/socketpool.c [hunk 2]
```diff
@@ -82,36 +84,49 @@ static void destroy_attached(struct pool_entry_t *entry)
...
-        if (entry->added_at > expire_before)
-            break;
-        destroy_attached(entry);
-        __sync_sub_and_fetch(&pool->_shared.count, 1);
-        __sync_sub_and_fetch(&pool->_shared.pooled_count, 1);
+/* caller should lock the mutex */
+static void check_pool_expired_locked(h2o_socketpool_t *pool)
+{
+    uint64_t next_expired = destroy_expired_locked(pool);
+    if (next_expired != SIZE_MAX) {
```
Comment (kazuho) [inline lib/common/socketpool.c]: Same here.

File: lib/common/socketpool.c (hunk 1 above)
Comment (chenbd) [reply lib/common/socketpool.c]: Yes, should be `UINT64_MAX` for 32 bit system.

File: lib/common/socketpool.c (hunk 2 above)
Comment (chenbd) [reply lib/common/socketpool.c]: Same as above.

Comment (chenbd) [reply lib/common/socketpool.c]: Same as above.

**Comment by kazuho** [comment]
Thank you for the changes.

**Pull request merged**